public_html/app/tools/Validator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+class Validator
+{
+
+    public $errors = [];
+
+    public function __construct()
+    {
+        $this->controller = new Controller();
+        $this->userModel = $this->controller->model('User');
+    }
+
+
+    public function validate($data, $validateData = [])
+    {
+        foreach ($validateData as $item => $rules) {
+            $this->passes($data[$item], $rules, $item);
+        }
+
+        // If errors[] consist of empty errors ' ', return true
+        $noErrors = (count(array_unique($this->errors)) === 1);
+
+        if ($noErrors) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public function passes($value, $rules, $fieldName)
+    {
+        foreach ($rules as $rule) {
+            $result = $this->{$rule}($value, $fieldName);
+
+            if ($result) {
+                $this->errors[] = '';
```

Comment:
you dont need append empty strings to array. simple
`if(!$result){$his->errors[] = $errorMessage; continue;)}`